### PR TITLE
ARC-2112 add logs for 405

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
Add logs to see
1. Why I couldn't find status calls failured at 302 on 405 request
2. Help decide future should we just wrap 405 as 200 or make it to 404 and delete message?

**Why**

**Added feature flags**

**Affected issues**  
ARC-2112

**How has this been tested?**  
Just logs

**Whats Next?**
Fix for all jira api endpoints on 405 (and potential other status)

### CODE REVIEW

File: src/jira/client/axios.ts [hunk 1]
```diff
@@ -166,13 +166,15 @@ const instrumentFailedRequest = (baseURL: string, logger: Logger) => {
 			// TODO: pretty sure this whole function is some dead code that we can delete, because in getErrorMiddleware
 			// TODO: the error is mapped to JiraClientError. If that's the case, the log line below should never be logged
 			// TODO: and we are safe to kill the thing.
-			logger.info("Ok, looks like still works.");
+			logger.info({ error }, "Ok, looks like still works.");
```
Comment (bgvozdev) [inline src/jira/client/axios.ts:169]: should be `{ err }`

File: src/jira/client/axios.ts [hunk 2]
```diff
@@ -183,6 +185,8 @@ const instrumentFailedRequest = (baseURL: string, logger: Logger) => {
 				}
 			}
 		}
+
+		logger.info({ jiraHost: baseURL, error }, "Rejecting final error");
```
Comment (bgvozdev) [inline src/jira/client/axios.ts:189]: `{ , err: error }`

**Review by bgvozdev — CHANGES_REQUESTED** [review]
we shouldn't log UGC/PII

File: src/jira/client/axios.ts (hunk 1 above)
Comment (gxueatlassian) [reply src/jira/client/axios.ts]: Good catch! 
Fixed now.

File: src/jira/client/axios.ts [hunk 3]
```diff
@@ -172,17 +173,22 @@ const instrumentFailedRequest = (baseURL: string, logger: Logger) => {
...
 					error.response.status = 404;
 				} else if (e.response.status === 302) {
-					logger.info({ jiraHost: baseURL },"405 from Jira: Jira instance has been renamed. Returning 404 to our application.");
+					logger.info("405 from Jira: Jira instance has been renamed. Returning 404 to our application.");
 					error.response.status = 404;
 				}
 			}
 		}
+
+		logger.info({ errorStatus: error.response?.status }, "Rejecting final error");
```
Comment (bgvozdev) [inline src/jira/client/axios.ts:191]: logging just `{ err }` would've worked fine btw :) in logger config we are removing all possible UGCs

**Review by bgvozdev — APPROVED** [review]

**Pull request merged**